### PR TITLE
Feature/vfb new layout2 canvas fixes

### PR DESCRIPTION
**Pull request by jrmartin**
- [ ] Add coverage for whatever new functionality, to a JUnit test if it's backend, to a Casper Test if it's frontend
- [ ] Make sure both push and pr travis builds are passing before asking for a review

### CODE REVIEW

File: src/main/webapp/js/components/interface/3dCanvas/ThreeDEngine.js [hunk 1]
```diff
@@ -137,126 +137,115 @@ define(['jquery'], function () {
...
-            	}
-            }, false);
-            
-            this.renderer.domElement.addEventListener('mouseup', function (event) {
-            	if (event.target == that.renderer.domElement) {
-            		var x = event.clientX;
-            		var y = event.clientY;
-
-            		// If the mouse moved since the mousedown then don't consider this a selection
-            		if( x != clientX || y != clientY )
```
Comment (ddelpiano) [inline src/main/webapp/js/components/interface/3dCanvas/ThreeDEngine.js]: This condition is quite critical to have as per first comment, differently we will end-up picking continuosly an item in the scene even when we want just rotate.

File: src/main/webapp/js/components/interface/3dCanvas/ThreeDEngine.js [hunk 2]
```diff
@@ -137,126 +137,115 @@ define(['jquery'], function () {
...
-            			return;
-
-
-            		that.mouse.x = (event.clientX / window.innerWidth) * 2 - 1;
```
Comment (ddelpiano) [inline src/main/webapp/js/components/interface/3dCanvas/ThreeDEngine.js]: this should be re-adapted as per mousemove event done in the new branch to get the value when the canvas is full screen or in a sub-window.

**Comment by ddelpiano** [comment]
@jrmartin this is great, I have few points to raise after testing:

- using your code without any change (since I was playing a bit with it later on) I noticed that the issue related to the canvas responsiveness fixed recently is now back, when I try to rotate the brain I can see that this takes a bit to react and then start to spin before to follow my movements.

- As per other comments into the code, we need to put in again the listener for the event mouseup and put again in place the logic for the selection in there and not in mousedown, differently we will end up selecting a different item in the scene every time we try to rotate the brain.
To reproduce this problem when you rotate try to pick the brain, **if you click outside the scene an empty space the rotation will work fine without delay.**

File: src/main/webapp/js/components/interface/3dCanvas/ThreeDEngine.js [hunk 3]
```diff
@@ -137,126 +137,115 @@ define(['jquery'], function () {
...
+            				that.deselectAll();
             			}
             		}
             	}
             }, false);
 
+
+            this.renderer.domElement.addEventListener('mousemove', function (event) {
+            	//canvas rendered inside whole HTML document
+            	if ( that.renderer.domElement === document ) {
```
Comment (ddelpiano) [inline src/main/webapp/js/components/interface/3dCanvas/ThreeDEngine.js]: This if - else I think can be removed and we should be able to take the that.mouse.y and that.mouse.x into the the else as valid in both the scenarios.

e.g.:
that.mouse.y = -((event.clientY -(window.innerHeight -that.renderer.domElement.height))/ that.renderer.domElement.height) * 2 + 1;

if canvas is fullscreen this -(window.innerHeight -that.renderer.domElement.height) should return 0.
Also we use that.renderer.domElement.height to divide the event.clientY, still if canvas is fullscreen this should be equal to window.innerHeight.

**Comment by ddelpiano** [comment]
@jrmartin I just pushed few changes related to the comments I posted before, this works fine for what I can see with both canvas in fullscreen or window.
I moved the logic you implemented for mouseup into mousedown and made few minor changes as per my comments, let me know what you think and in case feel free to revert if you need.

**Comment by jrmartin** [comment]
@ddelpiano Looks good, tested your changes and all works.

**Pull request merged**